mysqlnd_azure.c

```diff
@@ -610,6 +675,15 @@ MYSQLND_METHOD(mysqlnd_azure, connect)(MYSQLND * conn_handle,
                         const MYSQLND_CSTRING socket_or_pipe,
                         unsigned int mysql_flags)
 {
+  int logLevel = MYSQLND_AZURE_G(logLevel);
+  if (logLevel > 0) {
+    char *logfilePath = ZSTR_VAL(MYSQLND_AZURE_G(logfilePath));
+    OPEN_LOGFILE(logfilePath);
+  }
+
+  if (logLevel > 2) {
+    AZURE_LOG("DEBUG", "mysqlnd_azure.c: mysqlnd_azure::connect()");
```

Comment:
mysqlnd_azure.enableRedirect has unique value for same connection phase. So instead of log for every possible option value each time, we may log it only once outside the ifs/elses

Reply:
We print this information only when `mysqlnd_azure.logLevel >= 3` (very verbosely, ERROR, INFO, DEBUG).
If `mysqlnd_azure.logLevel == 1`, we only log ERROR level message(error that trigger a connection break);  when `mysqlnd_azure.logLevel == 2`, ERROR + INFO are logged, INFO means redirect connection fallbacks and cache usage information.

If someone changed this value, we can still find the context when connection broken.

If we only log it once, we cannot specify where the context came from under highly-currenct usage.

Reply:
Don't really understand you point here.. Maybe there is misunderstanding. What I mean here is only about logLevel 2, those three parts
if (logLevel > 2) {

                    AZURE_LOG("DEBUG", "mysqlnd_azure.enableRedirect: PREFERRD/ON/OFF");
can be stated outside the if/else only once. I understand this as they are in equivalent position in any log level

Reply:
There will be at  most one mysqlnd_azure.enableRedirect value logged per connection, because there is at most one fallback/connection break happen inside a connection. It seems number of logs will no more than we log it at every entry of mysqlnd_azure::connect().

Also, we only need to get that value when unexpected events happen. I dont think we should log the value down every time call mysqlnd_azure::connect().

There is a  huge interval between function entry and exception position., we cannot confirm whether the `enabledRedirection = XX` near a `some exception occurred` belong to a same connection or not.

Reply:
If one file will log for multi processes, then it is true we cannot make sure. But what I means here is not to only print out once, I just suggest log it for if level2 once outside the if/else, only for code clearance..